src/components/general-table/BulkSelect.js

```diff
@@ -7,19 +7,58 @@ import {
   ToolbarItem,
 } from '@patternfly/react-core';
 import PropTypes from 'prop-types';
+import useFetchBatched from '../../hooks/useFetchBatched';
+import { getInventory } from '../../api/devices';
+import { flatten, map } from 'lodash';
 
 const BulkSelect = ({
   checkedRows,
   handleBulkSelect,
   handlePageSelect,
   handleNoneSelect,
-  displayedRowsLength,
   perPage,
   total,
+  filters,
+  filterParams,
+  apiFilterSort,
 }) => {
-  const isAllSelected = checkedRows.length === displayedRowsLength;
-  const isPartiallySelected = checkedRows.length > 0 ? null : false;
+  const isAllSelected = checkedRows.length === total;
+  const isPartiallySelected = checkedRows.length > 0 ? true : false;
   const [selectAllToggle, setSelectAllToggle] = useState(false);
+  const { fetchBatched } = useFetchBatched();
+
+  const fetchAllSystemIds = useCallback((filters, total) => {
+    console.log(filters);
+    const query = apiFilterSort
+      ? {
+          ...filterParams(filters.filters),
+        }
+      : null;
+    return fetchBatched(getInventory, total, query);
```

Comment:
just to make sure I understood correctly
will this make multiple requests page by page to the edge-api to retrieve all the ids?

Reply:
explains during daily